app/experimenter/experiments/tests/api/v5/test_mutation.py

```diff
@@ -268,6 +289,52 @@ def test_update_experiment_error(self):
         )
         self.assertEqual(result["status"], 200)
 
+    def test_update_experiment_documentation_links(self):
+        user_email = "user@example.com"
+        experiment = NimbusExperimentFactory.create(status=NimbusExperiment.Status.DRAFT)
+        experiment_id = experiment.id
+
+        documentation_links = [
+            {"title": "foo", "link": "https://example.com/bar"},
+            {"title": "baz", "link": "https://example.com/quux"},
+            {"title": "xyzzy", "link": "https://example.com/plotz"},
+        ]
+
+        response = self.query(
+            UPDATE_EXPERIMENT_DOCUMENTATION_LINKS_MUTATION,
+            variables={
+                "input": {
+                    "id": experiment.id,
+                    "clientMutationId": "randomid",
+                    "documentationLinks": documentation_links,
+                }
+            },
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+        result = content["data"]["updateExperiment"]
+        self.assertEqual(
+            result["nimbusExperiment"],
+            {
+                "id": experiment.id,
+                "status": experiment.status.upper(),
+                "name": experiment.name,
+                "slug": experiment.slug,
+                "documentationLinks": documentation_links,
+            },
+        )
+
+        experiment = NimbusExperiment.objects.get(id=experiment_id)
+        experiment_links = experiment.documentation_links.all()
+        self.assertEqual(len(experiment_links), len(documentation_links))
+        for idx in range(0, len(documentation_links)):
+            link_data = documentation_links[idx]
+            link = experiment_links[idx]
+            for key, val in link_data.items():
+                self.assertEqual(getattr(link, key), val)
```

Comment:
Once you know the length is equal you can just check membership of each dict rather than looking them up by index.

Reply:
Switching to a set assertion here, which seems to rope everything in more concisely